### PR TITLE
Sort Sheets Alphabetically by vendorID

### PR DESCRIPTION
**Pull request by cxfloeder**
I implemented and tested code that sorts the Vendor and Account sheets alphabetically by vendorID.

### CODE REVIEW

File: src/main/java/data/SheetsConverter.java [hunk 1]
```diff
@@ -61,59 +62,69 @@ public void writeToSheet(ArrayList<Vendor> vendorList, Sheets sheetsService)
...
-        .getAccountSheetsRow(entry.getKey())));
+ 
+    // Sort the Account Sheet aphabetically by VendorID.
+    ArrayList<String> sortedKeys = sortKeys(accounts);
+ 
+    for(int i = 0; i < sortedKeys.size(); i++) {
+      ArrayList<Account> accountList = accounts.get(sortedKeys.get(i));
+      for(int j = 0; j < accountList.size(); j++) {
+        accountSheetData.add(new ArrayList<Object>(accountList.get(j)
+          .getAccountSheetsRow(sortedKeys.get(i))));
```
Comment (cadecrandall) [inline src/main/java/data/SheetsConverter.java:102]: Not a Java expert, but did you consider using something like a LinkedHashMap or TreeMap to be able to order entries? Building an entirely new map doesn't seem too efficient here

Comment (cxfloeder) [reply src/main/java/data/SheetsConverter.java:102]: I don't build a new map here. I use the existing map that gets passed into the method

File: src/main/java/data/SheetsConverter.java [hunk 2]
```diff
@@ -61,59 +62,70 @@ public void writeToSheet(ArrayList<Vendor> vendorList, Sheets sheetsService)
...
     }
-
+ 
     return allAccounts; 
   }
-
+ 
   public List<List<Object>> buildAccountSheetBody(
-      HashMap<String, ArrayList<Account>> accounts) {
+      HashMap<String, ArrayList<Account>> vendorIdMap) {
```
Comment (ianjjohnson) [inline src/main/java/data/SheetsConverter.java:91]: nit: maybe call this accountsByVendorId?

File: src/main/java/data/SheetsConverter.java [hunk 3]
```diff
@@ -61,59 +62,70 @@ public void writeToSheet(ArrayList<Vendor> vendorList, Sheets sheetsService)
...
-    accountSheetData.add(new ArrayList<Object>(Account.getAccountSheetHeader()));
-
-    for(Map.Entry<String, ArrayList<Account>> entry : accounts.entrySet()) { 
-      ArrayList<Account> accountList = entry.getValue();
-      for(int i = 0; i < accountList.size(); i++) {
-        accountSheetData.add(new ArrayList<Object>(accountList.get(i)
-        .getAccountSheetsRow(entry.getKey())));
+ 
+    // Sort the Account Sheet aphabetically by VendorID.
+    ArrayList<String> sortedKeys = sortKeys(vendorIdMap);
```
Comment (ianjjohnson) [inline src/main/java/data/SheetsConverter.java:96]: sortedVendorIds?

File: src/main/java/data/SheetsConverter.java [hunk 4]
```diff
@@ -61,59 +62,70 @@ public void writeToSheet(ArrayList<Vendor> vendorList, Sheets sheetsService)
...
       }
     }
 
+    accountSheetData.add(0, new ArrayList<Object>(Account.getAccountSheetHeader()));
+ 
     return accountSheetData;
   }
-
+ 
+  public ArrayList<String> sortKeys(HashMap<String, ArrayList<Account>> vendorIdMap) {
```
Comment (ianjjohnson) [inline src/main/java/data/SheetsConverter.java:111]: I would just inline this as Collections.sort(vendorIdMap.keySet()) and remove this method

File: src/main/java/data/SheetsConverter.java [hunk 5]
```diff
@@ -61,59 +62,70 @@ public void writeToSheet(ArrayList<Vendor> vendorList, Sheets sheetsService)
...
     List<List<Object>> accountSheetData = new ArrayList<List<Object>>();
-    accountSheetData.add(new ArrayList<Object>(Account.getAccountSheetHeader()));
-
-    for(Map.Entry<String, ArrayList<Account>> entry : accounts.entrySet()) { 
-      ArrayList<Account> accountList = entry.getValue();
-      for(int i = 0; i < accountList.size(); i++) {
-        accountSheetData.add(new ArrayList<Object>(accountList.get(i)
-        .getAccountSheetsRow(entry.getKey())));
+ 
+    // Sort the Account Sheet aphabetically by VendorID.
```
Comment (ianjjohnson) [inline src/main/java/data/SheetsConverter.java:95]: nit: spelling

File: src/main/java/data/Vendor.java [hunk 6]
```diff
@@ -24,7 +24,7 @@
 import java.util.stream.Collectors;
 
 /** A class representing a billing vendor object. */
-public class Vendor {
+public class Vendor implements Comparable<Vendor> {
```
Comment (ianjjohnson) [inline src/main/java/data/Vendor.java:27]: nice :-)

File: src/main/java/data/SheetsConverter.java (hunk 4 above)
Comment (cxfloeder) [reply src/main/java/data/SheetsConverter.java:111]: Sounds good

**Pull request merged**